packages/core/scripts/severless-cloud-run/CloudRunHub.js

```diff
@@ -88,120 +88,145 @@ app.post("/", async (req, res) => {
+        // correctly but contained an error.
+        errorOutputs.push({
           status: result.status,
-          execResponse: result.reason.response.data,
+          execResponse: result.value.execResponse,
           botIdentifier: Object.keys(configObject)[index]
         });
-      } else if (result.value.execResponse.error) {
-        // If the child process exited correctly but contained an error.
-        thrownErrors.push({
+      } else {
+        validOutputs.push({
           status: result.status,
           execResponse: result.value.execResponse,
           botIdentifier: Object.keys(configObject)[index]
         });
       }
     });
-    if (thrownErrors.length > 0) {
-      throw thrownErrors;
+    if (errorOutputs.length > 0) {
+      throw { errorOutputs, validOutputs };
     }
 
     // If no errors and got to this point correctly then return a 200 success status.
-    Logger.debug({
-      at: "CloudRunnerHub",
+    logger.debug({
+      at: "CloudRunHub",
       message: "All calls returned correctly",
-      thrownErrors: null
+      output: { errorOutputs, validOutputs }
+    });
+    res.status(200).send({ message: "All calls returned correctly", output: { errorOutputs, validOutputs } });
+  } catch (errorOutput) {
+    logger.error({
+      at: "CloudRunHub",
+      message: "Some calls returned errors 🚨",
+      output: errorOutput instanceof Error ? errorOutput.message : errorOutput
     });
-    res.status(200).send({ message: "All calls returned correctly", error: null });
-  } catch (thrownErrors) {
-    Logger.error({
-      at: "CloudRunnerHub",
-      message: "CloudRunner hub error 🌩 ",
-      thrownErrors
+    await waitForLogger(logger);
+
+    res.status(500).send({
+      message: "Some calls returned errors",
+      output: errorOutput instanceof Error ? errorOutput.message : errorOutput
     });
-    res.status(400).send({ message: "Something went wrong in cloud runner hub execution", thrownErrors });
   }
 });
 
-// Fetch a `file` from a GCP `bucket`. This function uses a readStream which is converted into a buffer such that the
-// config file does not need to first be downloaded from the bucket. This will use the local service account.
-const _fetchConfigObject = async (bucket, file) => {
-  const requestPromise = new Promise((resolve, reject) => {
-    let buf = "";
-    storage
-      .bucket(bucket)
-      .file(file)
-      .createReadStream()
-      .on("data", d => (buf += d))
-      .on("end", () => resolve(buf))
-      .on("error", e => reject(e));
-  });
+// Execute a CloudRun Post command on a given `url` with a provided json `body`. This is used to initiate the spoke
+// instance from the hub. If running in gcp mode then local service account must be permissioned to execute this command.
+const _executeCloudRunSpoke = async (url, body) => {
+  if (hubConfig.mode == "gcp") {
```

Comment:
Here the script defines if it's running in `gcp` or in `localStorage` mode. For now these are the only two options. Each of the main functions for executing a spoke, fetching a config and storing last block numbers have this if/else logic depending on the mode. In the future we can add additional modes for storing configs in github for example.